spec/support/ansible_shared/provider.rb

```diff
@@ -36,6 +36,46 @@
       expect(OperatingSystem.count).to       eq(0)
       expect(Hardware.count).to              eq(0)
     end
+
+    it "thru automation_manager will remove all child objects" do
+      subject.automation_manager.configured_systems = [
+        FactoryGirl.create(:configured_system, :computer_system =>
```

Comment:
you should use different factory https://github.com/Ladas/manageiq/blob/2f0ef1a90758f2b5c37e93a0ed2f586a916dad64/spec/factories/configured_system.rb#L10

or the default factory should contain valid base class STI, in this case ManageIQ::Providers::AutomationManager::ConfiguredSystem

-----

None of factories should be producing a record that can't be created in real life, e.g. ConfiguredSystem with type nil is not valid, refresh always set :type

Reply:
Thanks @Ladas !